fluent-plugin-events/lib/fluent/plugin/in_events.rb

```diff
@@ -138,33 +139,32 @@ def start_watcher_thread
       end
 
       def create_config_map
-        @resource_version = 0
-        @configmap.data = { "resource-version": "#{@resource_version}" }
+        @configmap.data = { "resource-version-#{resource_name}": "#{@resource_version}" }
         @client.public_send("create_config_map", @configmap).tap do |map|
           log.debug "Created config map: #{map}"
         end
       end
 
-      def update_config_map
+      def patch_config_map
         pull_resource_version
-        @configmap.data = { "resource-version": "#{@resource_version}"}
-        @client.public_send("update_config_map", @configmap).tap do |map|
-          log.debug "Updated config map: #{map}"
+        @client.public_send("patch_config_map", "fluentd-config-resource-version", {data: { "resource-version-#{resource_name}": "#{@resource_version}"}}, @namespace).tap do |map|
+          log.debug "Patched config map: #{map}"
         end
       end
 
       def initialize_resource_version
+        @resource_version = 0
         @configmap = ::Kubeclient::Resource.new
         @configmap.metadata = {
           name: "fluentd-config-resource-version",
-          namespace: "sumologic"
+          namespace: @namespace
```

Comment:
Correct me if I'm wrong, but I thought `@namespace` was for which namespace we want to watch resources from? I think we might need a separate parameter for which namespace we want to deploy in

Reply:
ah I think you are right.. I saw the namespace param and immediately thought it's the one we can use 😂